src/Perlang.Common/Attributes/GlobalClassAttribute.cs

```diff
@@ -23,5 +24,22 @@ public class GlobalClassAttribute : Attribute
         /// Gets or sets an optional custom name for the class. If not provided, the short-name of the class will be used.
         /// </summary>
         public string Name { get; set; }
+
+        /// <summary>
+        /// Gets a collection of all the platforms on which the class should be registered. If empty, the class should
+        /// be registered on all platforms.
+        /// </summary>
+        public IEnumerable<PlatformID> Platforms { get; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="GlobalClassAttribute"/> class.
+        /// </summary>
+        /// <param name="platforms">An optional list of platform IDs for which the given global class should be
+        /// registered. If no platform IDs are provided, the global class is presumed to be platform-agnostic and
+        /// registered on all supported platforms.</param>
+        public GlobalClassAttribute(params PlatformID[] platforms)
+        {
+            Platforms = platforms;
+        }
```

Comment:
Note that this approach has one major advantage: the code for a particular unsupported platform will still be _included_ (although not _active_) on that platform. This might not be a huge problem at first, but eventually we might want to be able to exclude it from the `.dll` altogether.

One way around it which would be possible right now would be to use `#if` directives to conditionally exclude this on particular platforms. This has the disadvantage of making it hard/impossible to work on this code on the unsupported platform though (the IDE will show this part of the code as "excluded"), so... not wanting to go that route for this particular reason.